src/main/host/descriptor/epoll.c

```diff
@@ -223,20 +251,35 @@ static void _epoll_free(LegacyDescriptor* descriptor) {
 void epoll_clearWatchListeners(Epoll* epoll) {
     MAGIC_ASSERT(epoll);
 
+    /* Iterate the hash table in a determinsitic order. */
+    GList* watch_list = g_hash_table_get_values(epoll->watching);
```

Comment:
Would it be better to use a data structure that maintains an ordering rather than sorting any time we need to iterate? e.g. https://developer.gnome.org/glib/2.68/glib-Balanced-Binary-Trees.html

Reply:
Added a note to consider trees. Since this PR appears to have a negligible performance impact, let's leave enhancements to epoll etc. for a future PR (maybe when we rewrite in rust).